### PR TITLE
support chinese indexing and search

### PR DESCRIPTION
**Pull request by Impavidity**
- Add CJKAnalyzer in the Indexing class.
- Add `language`  argument in indexing argument.
- Add `setLanguage` method and CJKAnalyzer in SimpleSearch class.
- Add experiments on NTCIR-8 ZH dataset.

TODO:

- Add more langugage support. 
- The language argument will follow ISO-639-2 code.

### CODE REVIEW

File: src/main/java/io/anserini/index/IndexCollection.java [hunk 1]
```diff
@@ -656,10 +656,12 @@ public void run() throws IOException {
     if (indexPath != null && !args.dryRun) {
 
       final Directory dir = FSDirectory.open(indexPath);
+      final CJKAnalyzer chineseAnalyzer = new CJKAnalyzer();
       final EnglishStemmingAnalyzer analyzer = args.keepStopwords ?
           new EnglishStemmingAnalyzer(args.stemmer, CharArraySet.EMPTY_SET) : new EnglishStemmingAnalyzer(args.stemmer);
       final TweetAnalyzer tweetAnalyzer = new TweetAnalyzer(args.tweetStemming);
-      final IndexWriterConfig config = args.collectionClass.equals("TweetCollection") ? new IndexWriterConfig(tweetAnalyzer) : new IndexWriterConfig(analyzer);
+      final IndexWriterConfig config = args.collectionClass.equals("TweetCollection") ?
+          new IndexWriterConfig(tweetAnalyzer) : args.language.equals("zh") ? new IndexWriterConfig(chineseAnalyzer) : new IndexWriterConfig(analyzer);
```
Comment (r-clancy) [inline src/main/java/io/anserini/index/IndexCollection.java]: Maybe this should be a normal nested if statement to improve readability?

File: src/main/java/io/anserini/index/IndexArgs.java [hunk 2]
```diff
@@ -76,6 +76,9 @@
   @Option(name = "-bm25.accurate", usage = "Switch to use the accurate BM25 similarity)")
   public boolean bm25Accurate = false;
 
+  @Option(name = "-language", usage = "boolean to use CJK analyser while indexing")
```
[the remark this answers is not in the document]
Comment (lintool) [reply src/main/java/io/anserini/index/IndexArgs.java]: Either actually implement the ISO-639-2 codes or change the option to, say, `-useCJK`?

File: src/main/java/io/anserini/index/IndexCollection.java (hunk 1 above)
Comment (lintool) [reply src/main/java/io/anserini/index/IndexCollection.java]: +1

**Comment by lintool** [comment]
@Impavidity also, Travis failed...

File: src/main/java/io/anserini/search/topicreader/TsvStringKeyTopicReader.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * 174249 does xpress bet charge to deposit money in your account
+ * 320792 how much is a cost to run disneyland
+ * 1090270  botulinum definition
+ * 1101279	do physicians pay for insurance from their salaries?
+ * 201376 here there be dragons comic
+ * 54544  blood diseases that are sexually transmitted
+ * ...
+ * </pre>
+ */
+public class TsvStringKeyTopicReader extends TopicReader<String> {
```
Comment (lintool) [inline src/main/java/io/anserini/search/topicreader/TsvStringKeyTopicReader.java]: This is mostly duplicate with https://github.com/castorini/anserini/blob/master/src/main/java/io/anserini/search/topicreader/TsvTopicReader.java

Seems like the only different is either treating the key as an int or the key as a string...

I don't know if the distinction is important, but let's keep it for now.

So, refactor: `TsvTopicReader` -> `TsvIntTopicReader`, this class to `TsvStringTopicReader`?

File: src/main/resources/topics-and-qrels/topics.ntcir8zh.eval.txt [hunk 4]
```diff
@@ -0,0 +1,73 @@
+ACLIA2-CS-0002	《千里走单骑》和张艺谋是什么关系？
```
Comment (lintool) [inline src/main/resources/topics-and-qrels/topics.ntcir8zh.eval.txt]: Can we make the qrels and topics have the same name? Why `ntcir8zh` vs. `nctir8` above?

File: src/main/python/clir/convert_collection_to_jsonl.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
```
Comment (lintool) [inline src/main/python/clir/convert_collection_to_jsonl.py]: Why do we need this?
The collection is in TREC doc format... the existing collection should be able to parse it?
We should avoid rewriting collections unless absolutely necessary...

File: src/main/java/io/anserini/search/topicreader/TsvStringKeyTopicReader.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+/**
+ * Topic reader for queries in tsv format, such as the MS MARCO queries.
```
Comment (lintool) [inline src/main/java/io/anserini/search/topicreader/TsvStringKeyTopicReader.java]: Please fix the comments.

File: docs/experiments-clir.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
+# Cross-lingual Information Retrieval Experiments
```
Comment (lintool) [inline docs/experiments-clir.md]: See https://github.com/castorini/anserini/blob/master/src/main/resources/docgen/templates/robust04.template for an example of how this is actually done.

Please make consistent?

Title should be collection-specific (e.g., NTCIR).

**Comment by lintool** [comment]
@Impavidity bunch of comments for you to address... since we're going to build on this foundation, let's get it right...

File: src/main/java/io/anserini/search/topicreader/TsvStringKeyTopicReader.java (hunk 3 above)
Comment (Impavidity) [reply src/main/java/io/anserini/search/topicreader/TsvStringKeyTopicReader.java]: Any reason we want to use int instead of string?

Comment (lintool) [reply src/main/java/io/anserini/search/topicreader/TsvStringKeyTopicReader.java]: I think it's the sort order? 100 sorts before 99 based on String.

Actually, you know what? Refactor into a link list map, and let's just preserve whatever order comes in the topic file.

That way we don't need two separate implementations.

The key can be String in this case.

What do you think?

File: src/main/python/clir/convert_collection_to_jsonl.py (hunk 5 above)
Comment (Impavidity) [reply src/main/python/clir/convert_collection_to_jsonl.py]: Is this trec doc format ?
<DOC>
<DOCNO> XIN_CMN_20020101.0001 </DOCNO>
    新华社一月一日发稿目录(二)
</DOC>

**Pull request merged**